keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactVersionLinks.kt

```diff
@@ -51,22 +53,41 @@ class ArtifactVersionLinks(private val scmInfo: ScmInfo, private val cacheFactor
     } else null
   }
 
+  fun getScmType(commitLink: String): String? {
+    val commitURL = URL(commitLink)
+
+    val scmBaseURLs = runBlocking {
+      val cachedValue = cache[cacheName].get()
+      cachedValue ?: scmInfo.getScmInfo()
+    }
+
+    val base = scmBaseURLs.filter { (_,baseUrl) ->
+      commitURL.host == URL(baseUrl).host
+    }
+
+    return base.keys.toList().getOrNull(0)
```

Comment:
probably slightly easier if `getScmType` return the lowercase type.

```suggestion
    return base.keys.toList().getOrNull(0)?.let { it.lowercase() }
```

Reply:
right but per my earlier comment, the intent is less explicit even if is more dry. A compromise could be renaming getScmType to getNormalizedScmType or getLowerScmType, I like the former in case there is more work to normalize these strings in the future.

Reply:
yeah lets go with `getNormalizedScmType`. sounds good to me